### PR TITLE
Add names to repr for named tensors

### PR DESCRIPTION
**Pull request by zou3519**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23625 Implement tensor.align_to(names), torch.align_tensors(*tensors)
* #23624 Add name propagation for at::alias, add tensor.set_names
* **#23316 Add names to repr for named tensors**
* #23623 Add torch._C._BUILD_NAMEDTENSOR()

Adds a `names=` argument to the repr() of tensor if `tensor.has_names()`.

Test Plan:
- [namedtensor ci]

gh-metadata: pytorch pytorch 23316 gh/zou3519/80/head

Differential Revision: [D16494415](https://our.internmc.facebook.com/intern/diff/D16494415)

### CODE REVIEW

File: test/test_namedtensor.py [hunk 1]
```diff
@@ -85,6 +85,18 @@ def test_has_names(self):
         self.assertTrue(partially_named.has_names())
         self.assertTrue(fully_named.has_names())
 
+    def test_repr(self):
+        named_tensor = torch.zeros(2, 3).set_names_(['N', 'C'])
```
Comment (nairbv) [inline test/test_namedtensor.py]: It's not really related to this particular PR, but do we have other tensor functions named this way? To me the `set` and trailing `_` seem to redundantly express the same thing. I feel like we could have a better name for it.

Comment (zou3519) [reply test/test_namedtensor.py]: We have:
- set_data(...)
- set_
- set_quantizer_
so an example and a counter-example. I'm taking suggestions for a better name!

**Comment by nairbv** [comment]
I think it'd be clearer to name this commit message something like 'add names to tensor repr,' I got a bit confused thinking I'd see some newly implemented repr() function.

**Review by nairbv — APPROVED** [review]

**Pull request closed without merging**